### PR TITLE
Improved readability for glean schemas

### PR DESCRIPTION
**Pull request by annanixon**
Updated parser for improved readability of glean schemas

@dse0 looked into indenting but couldn't find a great way to do that in markdown so think it would be better to add CSS rules to the `<details>` elements from our doc repo

This PR is just for the updated markdown output, and the files in here are only examples. Will have a separate PR to pull this into our doc repo and generate docs from schemas in prod.

### CODE REVIEW

**Comment by hashboard-charlie** [comment]
I feel like I've lost the plot here. This is a new repo? Okay, and we've formatted the `index.js` file... OK. Can we have that formatting in a separate commit or something? Hard to see what you changed.

What's the plan for how these make their way into their destination? https://github.com/gleannyc/glean/pull/6056#pullrequestreview-1210800309
This PR seems to commit results (and inputs) into this repo... that can't be right?

**Comment by annanixon** [comment]
Sorry, should've added more context. This repo is a fork of the repo that was used in #6056, it'll be a dependency that will be used in the PR that goes out to documentation repo where the files will actually be generated by pulling from the production branch like you suggested. The results and inputs committed to this repo were just to test what the output would look like with our schema but they won't be used.

File: index.js [hunk 1]
```diff
@@ -143,7 +163,7 @@ function generateSchemaSectionText(
 					text = text.concat(
 						generateSchemaSectionText(
 							octothorpes,
-							item.title,
+							item.title ?? name.slice(0, -1),
```
Comment (hashboard-charlie) [inline index.js:166]: why are chopping off the last char of this name..?

**Review by hashboard-charlie — APPROVED** [review]
Thanks for splitting up the commits!

File: index.js (hunk 1 above)
Comment (annanixon) [reply index.js:166]: A lot of fields in our schemas (like globalFilters, blocks, etc) have unnamed child objects so the output looks bad. This is just to depluralize the parent name and use it for the child object if it's unnamed

**Pull request merged**